tests/priority_queue_test.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/priority_queue.h>
+#include <aws_test_harness.h>
+
+static int int_compare(void *a, void *b) {
+    int *a_val = (int*)a;
+    int *b_val = (int*)b;
+
+    return *a_val < *b_val;
+}
+
+static int test_priority_queue_preserves_order(struct aws_allocator *alloc, void *ctx) {
```

Comment:
There was a major bug in the implementation of pop, which suggests that the test case is not comprehensive enough. I would recommend adding a randomized test - say, maintain a set of 16 +- 4 elements, and randomly push/pop, verifying that the correct elements are returned by comparing against a very simple unsorted array that you scan to find the maximum.